### ISSUE TITLE
[Scala 3] Breakpoint within a match expression does not find local variables

### ISSUE THREAD

**Issue by julienrf** [scalacenter/scala-debug-adapter#241]
~~~ scala
package foo

def f(n: Int): Int =
  n match
    case m =>
      println("foo") // (1)
      m + m          // (2)

@main def run() =
  println(f(21))
~~~

If I put the breakpoint at `(1)` and try to evaluate `m`, it fails with a `NoSuchElementException: m`. If I put the breakpoint at `(2)`, it works as expected.

**Comment by adpi2** [comment]
Thank you @julienrf, it seems there is a bug in the compiler itself: https://github.com/lampepfl/dotty/issues/15535

**Comment by adpi2** [comment]
Fixed in Scala 3.2.1, as shown by the new test in https://github.com/scalacenter/scala-debug-adapter/pull/310
